### ISSUE TITLE
Possible to replace Ping with TCP or ARP scan to better iOS (iPhone) detection

### ISSUE THREAD

**Issue by GatoPharaoh** [PeteLawrence/homebridge-people#51]
I have some iOS-devices at home (mostly from my loved ones) that don't wake that often which means that they often turn to a false state even though they are in the network. Those are devices that mostly lie around when at home (because the are mainly used for calls) which means they wake pretty rarely.

It seems like iOS devices (mainly iPhones - which are most interesting in this use case) won't respond to ping if not waken regularly.

According to [this thread](https://www.reddit.com/r/jailbreak/comments/27v37a/wifi_turns_off_when_locked_how_do_notifications/ci4pvy1/), ping won't work if sleeping for a while but sending TCP packets will work.

Another possibility I saw was scanning the ARP table - described [here](http://blog.onefellow.com/post/129574200528/how-to-detect-iphone-in-your-local-network)

Are any of these easy to implement?

By the way I'm in no way an expert on this - just researched a lot why those devices that "lie around" the whole day, will often not respond to ping.

**Comment by benzman81** [comment]
I had the same issue with my wifes iphone. It seems to depend on some settings and apps when a device wakes up. That is why I changed the plugin to be able to set differen thresholds for each device.

**Comment by GatoPharaoh** [comment]
what threshold value did you choose? because at the moment (I'm stilling in testing though) it seems that those devices will need at least an hour to respond again to ping which makes it pretty impractical for automation (e.g. turn off the lights when leaving home etc.)

**Comment by jojost1** [comment]
> what threshold value did you choose?
Also interested in this!

**Comment by benzman81** [comment]
For now, I have 15min for my phone set and 21 min fpr my wife. But still needs adjustments, no time yet to investigate further.

**Comment by richie510** [comment]
I could only get ping to work for IP addresses on iPhones, so I set them up for static IP's. A threshold of 7 minutes worked 99% of the time for me, but 8 works 100%. I'm using an iphone 7 and wife has 7+. Not sure if the newer phones respond better.

**Comment by jojost1** [comment]
Does it matter (for iPhones) if they are connected to 2.4 or 5GHz networks?

**Comment by GatoPharaoh** [comment]
I have both frequencies operating and it doesn't make any difference for me

**Comment by GatoPharaoh** [comment]
So for anyone interested in ARP and UDP wake up, I found a reliable workaround.
The phones I had trouble with are now responding at any time. 

I took inspiration from the guys over at the openhab forum and wrote a script that uses hping3 which will wake the phone to let it respond to pinging.
The scripts itself triggers the corresponding webkook of homebridge-people depending on the presence of the phone/person.

If anyone is interested, I'll write up necessary steps.
